drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-backend/src/main/java/org/drools/workbench/screens/guided/dtable/backend/server/conversion/BRLColumnSubHeaderBuilderByPatterns.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.drools.workbench.screens.guided.dtable.backend.server.conversion;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Objects;
+
+import org.drools.workbench.models.datamodel.rule.ActionSetField;
+import org.drools.workbench.models.datamodel.rule.IAction;
+import org.drools.workbench.models.guided.dtable.shared.model.ActionCol52;
+import org.drools.workbench.models.guided.dtable.shared.model.BRLActionColumn;
+import org.drools.workbench.models.guided.dtable.shared.model.BRLActionVariableColumn;
+import org.drools.workbench.models.guided.dtable.shared.model.BRLConditionColumn;
+import org.drools.workbench.models.guided.dtable.shared.model.BRLConditionVariableColumn;
+import org.drools.workbench.models.guided.dtable.shared.model.ConditionCol52;
+import org.drools.workbench.models.guided.dtable.shared.model.GuidedDecisionTable52;
+import org.drools.workbench.screens.guided.dtable.backend.server.conversion.util.ColumnContext;
+import org.drools.workbench.screens.guided.dtable.backend.server.conversion.util.FromTo;
+
+import static org.drools.workbench.screens.guided.dtable.backend.server.conversion.util.Util.hasContent;
+
+/**
+ * Splits the BRL column so that each value has a column.
+ */
+public class BRLColumnSubHeaderBuilderByPatterns
+        implements BRLColumnSubHeaderBuilder {
+
+    private SubHeaderBuilder subHeaderBuilder;
+    private ColumnContext columnContext;
+
+    private GuidedDecisionTable52 dtable;
+
+    public BRLColumnSubHeaderBuilderByPatterns(final SubHeaderBuilder subHeaderBuilder,
+                                               final ColumnContext columnContext,
+                                               final GuidedDecisionTable52 dtable) {
+        this.subHeaderBuilder = subHeaderBuilder;
+        this.columnContext = columnContext;
+        this.dtable = dtable;
+    }
+
+    @Override
+    public void brlActions(final BRLActionColumn brlColumn) {
```

Comment:
It seems that this method is doing too many things. Could you please modularize it in multiple methods with proper names?